### PR TITLE
Add acquired singleton shared resources helper

### PR DESCRIPTION
**Pull request by gavin-norman-sociomantic**

### CODE REVIEW

File: src/swarm/neo/util/AcquiredResources.d [hunk 1]
```diff
@@ -372,6 +372,167 @@ unittest
 
 /*******************************************************************************
 
+    Singleton (per-request) acquired resources of the templated type. An
+    external source of elements of this type -- a FreeList!(T) -- is required.
+    When the singleton resource is acquired (via the acquire() method), it is
+    requested from the free list and stored internally. All subsequent calls to
+    acquire() return the same instance. When the resource is nolonger required,
```
Comment (nemanja-boric-sociomantic) [inline src/swarm/neo/util/AcquiredResources.d]: `nolonger` typo

File: src/swarm/neo/util/AcquiredResources.d [hunk 2]
```diff
@@ -372,6 +372,167 @@ unittest
 
 /*******************************************************************************
 
+    Singleton (per-request) acquired resources of the templated type. An
```
Comment (nemanja-boric-sociomantic) [inline src/swarm/neo/util/AcquiredResources.d]: It seems that my broken English can't parse `Singleton acquired resources of the templated type.`, but I'm not sure if this is incorrect.

File: src/swarm/neo/util/AcquiredResources.d [hunk 3]
```diff
@@ -372,6 +372,167 @@ unittest
...
+{
+    import ocean.util.container.pool.FreeList;
+
+    /***************************************************************************
+
+        Determine the type of a new resource.
+
+    ***************************************************************************/
+
+    static if ( is(typeof({T* t = new T;})) )
```
Comment (nemanja-boric-sociomantic) [inline src/swarm/neo/util/AcquiredResources.d:397]: Could you use `isReferenceType` from `Traits_tango` instead?

File: src/swarm/neo/util/AcquiredResources.d (hunk 2 above)
Comment (gavin-norman-sociomantic) [reply src/swarm/neo/util/AcquiredResources.d]: I suppose "singleton acquired resource" would be more accurate.

File: src/swarm/neo/util/AcquiredResources.d (hunk 3 above)
Comment (gavin-norman-sociomantic) [reply src/swarm/neo/util/AcquiredResources.d:397]: Yeah, I guess that would have the same result. I just copied what's being used already in this module (https://github.com/gavin-norman-sociomantic/swarm-tsunami/blob/9b63f1077d8ae0c2cfa64c270ed99124185f0a2d/src/swarm/neo/util/AcquiredResources.d#L201).

Comment (nemanja-boric-sociomantic) [reply src/swarm/neo/util/AcquiredResources.d:397]: Fair enough, do as you please, I'm fine with both.

**Comment by gavin-norman-sociomantic** [comment]
Updated with typos fixed.

**Review by nemanja-boric-sociomantic — APPROVED** [review]

**Pull request merged**